### PR TITLE
clean up unit-test errors when it timeouts

### PR DESCRIPTION
**Pull request by lanchongyizu**
**Backgroud**
This is for the story https://rackhd.atlassian.net/browse/RAC-5639.

During work of https://rackhd.atlassian.net/browse/RAC-5593, we found many unit test cases won't do the clean-up work if running into failure (aka exception handler) , so the plan is to add clean-up work in unit test cases.

**TEST**
It's tested in sandbox:
http://10.62.59.175:8080/job/on-core_jamie/

@RackHD/corecommitters

### CODE REVIEW

**Comment by anhou** [comment]
@lanchongyizu Could you help post the errors in the RAC or in this PR's message? I see lots of minor changes which is not related to exception handler.

File: lib/services/waterline.js [hunk 1]
```diff
@@ -125,8 +129,15 @@ function WaterlineServiceFactory(
         .then(function() {
             //it's OK to create indexes multiple times, so the code isn't protected by initialized
             //checking.
+            //
+            //Catching errors is because:
+            //  (1) When it fails to create indexes, the service still works.
```
Comment (anhou) [inline lib/services/waterline.js]: So does it mean that if the error is catched, and the service could continue to work?

File: spec/lib/common/arp-spec.js [hunk 2]
```diff
@@ -22,16 +22,13 @@ describe('ARPCache', function () {
     before(function () {
         arpCache = helper.injector.get('ARPCache');
         fs = helper.injector.get('fs');
-        sinon.stub(fs, 'readFileAsync');
     });
 
     beforeEach(function() {
-        fs.readFileAsync.reset();
+        this.sandbox.stub(fs, 'readFileAsync');
```
Comment (anhou) [inline spec/lib/common/arp-spec.js]: I see that readFileAsync 's stub, restore, reset handling is not aligned in this PR

**Comment by lanchongyizu** [comment]
The main errors are in SH sandbox Jenkins, such as http://10.62.59.175:8080/blue/organizations/jenkins/on-core_jamie/detail/on-core_jamie/35/tests

File: lib/services/waterline.js (hunk 1 above)
Comment (lanchongyizu) [inline lib/services/waterline.js]: Right.

**Pull request merged**